### PR TITLE
use-readines-probe

### PR DESCRIPTION
**Pull request by calvix**
We already have quite nice health-checks on the kvm-pod which are testing: ping, kubelet health endpoint and for master nodes, it also checks API availability.

Using same port for readiness probe make sense as if kubelet is not ready we should not consider kvm-pod ready too.

This helps us in kvm updates as kvm logic loop check if the deployment is satisfied and ready:
https://github.com/giantswarm/kvm-operator/blob/master/service/controller/v18/resource/deployment/update.go#L79-L89

That means if the node doesn't have kubelet up it won't be considered ready and update logic should not continue.

https://github.com/giantswarm/giantswarm/issues/5363

### CODE REVIEW

**Review by mazzy89 — APPROVED** [review]

File: service/controller/v18/key/key.go [hunk 1]
```diff
@@ -407,6 +407,10 @@ func PVCNames(customObject v1alpha1.KVMConfig) []string {
 	return names
 }
 
+func ReadinessPort(customObject v1alpha1.KVMConfig) int32 {
+	return int32(livenessPortBase + customObject.Spec.KVM.Network.Flannel.VNI)
```
Comment (teemow) [inline service/controller/v18/key/key.go]: Imo if this should use the same port as the liveness probe then you should actually call LivenessPort() in here.

**Review by teemow — APPROVED** [review]

**Comment by calvix** [comment]
maybe it's worth backporting into security patches so the updates are more smooth, WDYT?

**Comment by teemow** [comment]
@calvix yes backporting would be great. /cc @webwurst

**Comment by calvix** [comment]
backported, but just to be clear older version have the only ping as health/readiness checks so improvement won't be as effective as in latest release, but should help a little bit

**Comment by calvix** [comment]
run first testing round on dinosaur:
*  it nows definitely roll nodes more gracefully, but i feel its maybe too slow, the update loop
```
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/update.go:109","controller":"kvm-operator","event":"update","function":"NewUpdatePatch","level":"debug","loop":"36","message":"not updating deployment 'worker-4y0ku': no changes found","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.455323+00:00","version":"46739921"}
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/update.go:113","controller":"kvm-operator","event":"update","function":"NewUpdatePatch","level":"debug","loop":"36","message":"found deployment 'worker-7qr79' that has to be updated","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.455355+00:00","version":"46739921"}
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/create.go:39","controller":"kvm-operator","event":"update","function":"ApplyCreateChange","level":"debug","loop":"36","message":"the deployments do not need to be created in the Kubernetes API","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.455396+00:00","version":"46739921"}
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/delete.go:41","controller":"kvm-operator","event":"update","function":"ApplyDeleteChange","level":"debug","loop":"36","message":"the deployments do not need to be deleted from the Kubernetes API","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.455442+00:00","version":"46739921"}
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/update.go:26","controller":"kvm-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"36","message":"updating the deployments in the Kubernetes API","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.455495+00:00","version":"46739921"}
{"caller":"github.com/giantswarm/kvm-operator/service/controller/v18/resource/deployment/update.go:36","controller":"kvm-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"36","message":"updated the deployments in the Kubernetes API","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/kvmconfigs/fp544","resource":"deploymentv18","time":"2019-02-19T11:13:21.459697+00:00","version":"46739921"}
```
seems like it run every x minutes which  could be faster


*  the readiness probe needs its own initialDelay as its too long  if we use same for livenessProbe (fixed that)
*I will also port the new k8s-kvm-health to the security release

**Comment by calvix** [comment]
Looks good now,  the rolling of nodes is definitely much better than before

**Review by xh3b4sd — APPROVED** [review]
:shipit:

**Pull request merged**